README.md

```diff
@@ -1504,296 +1504,151 @@ Sources:
-   import { useContext } from "react";
-
-   const [ctx, TextProvider] = createCtx("someText");
-   export const TextContext = ctx;
-   export function App() {
-     return (
-       <TextProvider>
-         <Component />
-       </TextProvider>
-     );
-   }
-   export function Component() {
-     const { state, update } = useContext(TextContext);
-     return (
-       <label>
-         {state}
-         <input type="text" onChange={(e) => update(e.target.value)} />
-       </label>
-     );
-   }
-   ```
-
-   [View in the TypeScript Playground](https://www.typescriptlang.org/play/?jsx=2#code/JYWwDg9gTgLgBAJQKYEMDG8BmUIjgcilQ3wFgAoCpAD0ljkwFcA7DYCZuNIlGJAYRjUAPAEEAfAAoAJkkwpGAGxgA1FIsZIAXHFEBKOAG8KcODACeYJHACqYabyQAVS9YC8iYjAB0AEWAAzmC8aAAWwsjoPgDKSDDRMI6ibBzCFlYQmHCy8kqq6pri4gDcJlwcAfA5Csp2Dnw6dY4uVnAekgZu4tlyNfkaSKXkpmgV8BjUbZ5R3tyofPwcfNQwksbDpnCVjjrVeWoDADRlpoz2Oz25ted8ZQC+ekOmTKww7JwACjgAbsCyUJIwDgwAEdJEMN4vhAQQB1YAwUL8ULARTSIjMYSGO7iAzrTblZiVOAAbW2fEOcDO9SQAF0puCfIwAkgEo4ZL19gUkI8TnAiDBGFBOMIJpCfn8kFA4N8uW5DIYtolyZSbtY7ncjN4tUDoQENQB6Er3Mr8wWcYkTClQ37-OkoAIEyrFOD6-VwdR8IW8YDfJCKcwU4npJCZLhCCnB0PWiVQGkUO4UCiuykBFAAcyQifIo0J8At4bgThoMGjtqmc0cgmokgARAFcM5izWeeQaHRxmNC8XFsxlvAPBMhm3oFgWClOKIwGAOkYTXEzXBJLzhEWVqXJeJeaZhItwBwkL2XZuNtv9auS+L-sfTC2E63aCOGGO3hw4LvIMwD6tcWUc0SFWSSAUlSjhwBqHgMt4TICEsxaSOePZ9i2pimkKi7LooKAAEZ+te+JGIBd74XAwjAMwYCMPAwZuDWfY1nAHBIigzAZnK7jdCBfCSEg3iJFAGY+DKAx6AaeGnphOGKHht5AA)
-
-5. A [useReducer-based version](https://gist.github.com/sw-yx/f18fe6dd4c43fddb3a4971e80114a052) may also be helpful.
+```tsx
+import { createContext } from "react";
 
-<details>
+interface CurrentUserContextType {
+  username: string;
+}
+
+const CurrentUserContext = createContext<CurrentUserContextType | null>(null);
+
+const useCurrentUser = () => {
+  const currentUserContext = useContext(CurrentUserContext);
 
-<summary><b>Mutable Context Using a Class component wrapper</b></summary>
+  if (!currentUserContext) {
+    throw new Error(
+      "useCurrentUser has to be used within <CurrentUserContext.Provider>"
+    );
+  }
+
+  return currentUserContext;
+};
+```
 
-_Contributed by: [@jpavon](https://github.com/typescript-cheatsheets/react/pull/13)_
+Using a runtime type check in this will has the benefit of printing a clear error message in the console when a provider is not wrapping the components properly. Now it's possible to access `currentUser.username` without checking for `null`:
 
 ```tsx
-interface ProviderState {
-  themeColor: string;
-}
+import { useContext } from "react";
 
-interface UpdateStateArg {
-  key: keyof ProviderState;
-  value: string;
-}
+const MyComponent = () => {
+  const currentUser = useCurrentUser();
 
-interface ProviderStore {
-  state: ProviderState;
-  update: (arg: UpdateStateArg) => void;
-}
+  return <p>Username: {currentUser.username}.</p>;
+};
+```
 
-const Context = createContext({} as ProviderStore); // type assertion on empty object
+##### Type assertion as an alternative (not recommended)
```

Comment:
Let's not do this for now. There are instances where type assertions are better (since they have no runtime impact) but it's on a case-by-case basis.

The rest is perfect. Nice job 👍🏻

Reply:
Do you mean let's remove the "(not recommended)" part?

Reply:
Yeah. Though I think we should help somewhat in guiding. So maybe downgrade it from the heading to a paragraph like "when you don't know which to chose, use runtime checking and throwing".

Reply:
Added this in the end now:

> When you don't know what do choose, prefer runtime checking and throwing over type asserting.